core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -105,6 +108,8 @@ public void initialize(String name, Map<String, String> properties) {
     this.fileIO = fileIOImpl == null ? new HadoopFileIO(conf) : CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
 
     this.suppressPermissionError = Boolean.parseBoolean(properties.get(HADOOP_SUPPRESS_PERMISSION_ERROR));
+
+    this.lockManager = LockManagers.from(properties);
```

Comment:
Nit: Can you put this just after `fileIO` to match the order of the field declarations?

Reply:
done

Reply:
I don't think we should do this for `HadoopCatalog`, because hadoop catalog users usually can leverage Hadoop file system's mutual exclusive writing feature and does not need a pessimistic lock to do commit. I think we should allow `LockManager` to be null in `HadoopTableOperations`, and only if it is configured, we use it. 

This probably also means we need another config property `lock-enabled`.

Reply:
@jackye1995 I think users can still use HadoopCatalog with S3 or local file system even that's not the optimal solution, so the data loss issue is still there?

Reply:
@jackye1995, I don't think there's a problem using the lock manager by default, considering the implementation is in-memory only. That's not going to cause much overhead.